### DISCUSSION TITLE
Default key bindings for controllers

### DISCUSSION THREAD

**Discussion by Susko3** [ppy/osu#25573] [Ideas]
We have reports of [people trying to use a controller, but it not working](https://discord.com/channels/188630481301012481/1097318920991559880/1177720273642991696) since the game has no default keybinds for controllers.

The only thing that currently supports controller input of the the box is the initial osu! cookie (pressing any button will activate it).
I think it would be beneficial for discovery to support basic navigation with controllers (just basic up, down, left, right, select and back).

But just adding the bindings looks really bad in the settings screen:

![image](https://github.com/ppy/osu/assets/16479013/948aff6b-26c8-47ae-bdbc-9b8a89ddea95)

I'm not sure what's the best way to make this look good. Here are some (bad) ideas:
1. display bindings in two rows if they overflow
2. have a separate key configuration menu for controllers (I don't really like this one)
3. hide the bindings until a controller event is detected (used in combination with 1. to prevent overflow)

**Comment by Susko3** [comment]
I also had this fun idea about navigating the menu. (This can be added without adding anything to the settings screen.)

![image](https://github.com/ppy/osu/assets/16479013/396f7991-b604-4ae7-b419-08a35e453424)

**Reply by Susko3** [reply]
Not sure about hijacking the <kbd>A</kbd> and <kbd>B</kbd> buttons, as one expects those to be select (selects the cookie) and back (tho exit is matching, but it would prevent going to the initial cookie-only screen).

**Comment by Joehuu** [comment]
Relevant display issue: https://github.com/ppy/osu/issues/6526

**Comment by Susko3** [comment]
[Fontawesome `gamepad`](https://fontawesome.com/icons/gamepad?f=classic&s=solid) doesn't look half-bad. But it'll still overflow if there are more than 4 keybinds for an action (not pictured here).

![image](https://github.com/ppy/osu/assets/16479013/daf537b6-2395-4b38-9438-2383af4c53ff)

**Comment by peppy** [comment]
I think rather than polluting more defaults, there should be a screen in the first run wizard to set sane defaults for a specific input device. Controllers on osu! are so rare that I think it should be something the user needs to explicitly decide to change.
